mongosql/src/codegen/match_query.rs

```diff
@@ -29,6 +29,7 @@ impl MqlCodeGenerator {
             Regex(r) => self.codegen_match_regex(r),
             ElemMatch(em) => self.codegen_match_elem_match(em),
             Comparison(c) => self.codegen_match_comparison(c),
+            False => Ok(bson!({"_id": Bson::MinKey, "$expr": false})),
```

Comment:
```suggestion
            // Some versions of MongoDB do not properly optimize {$match: { $expr: false } } so we codegen it
            // this way to ensure efficient performance. Otherwise, such a query could result in a needless collection
            // scan.
            False => Ok(bson!({"_id": Bson::MinKey, "$expr": false})),
```
We'd also benefit from a comment here.